bench/workgen/workgen.i

```diff
@@ -174,6 +174,23 @@ WorkgenFrozenClass(WorkloadOptions)
         if config == None:
             config = self.default_config
         self.initialize()
+        /* 
+         * For tiered storage get the bucket directory name from the connection string and 
+         * create if does not exist.
+         */
+        bucketConfigIndex = config.find("bucket=")
+        if bucketConfigIndex != -1:
+            bucketString = config[bucketConfigIndex:]
```

Comment:
Unfortunately we cannot create bucket directories inside WT home from the benchmark script because WT home directory is created before wiredtiger_open [here](https://github.com/wiredtiger/wiredtiger/blob/develop/bench/workgen/workgen.i#L176). I modified the script to create bucket directories only in dir_store extension.

Reply:
I'm still not excited about the parsing solution, parsing is hard to get right, corner cases abound.  How about this (in workgen.i)?
```
  def wiredtiger_open(self, config = None, post_initialize_function = None):
        if config == None:
            config = self.default_config
        self.initialize()
        if post_initialize_function != None:
            post_initialize_function(self.args.home)
        return wiredtiger.wiredtiger_open(self.args.home, self.wiredtiger_open_config(config))
```
Then you can set the `post_initialize_function` to a function that creates the bucket directory.

Reply:
Thanks @ddanderson . This looks much better.